lldb/source/Host/posix/DomainSocket.cpp

```diff
@@ -182,3 +191,22 @@ std::vector<std::string> DomainSocket::GetListeningConnectionURI() const {
 
   return {llvm::formatv("unix-connect://{0}", addr.sun_path)};
 }
+
+Socket *DomainSocket::Create(NativeSocket sockfd, bool should_close,
+                             Status &error) {
+#ifdef __linux__
+  // Check if fd represents domain socket or abstract socket.
+  struct sockaddr_un addr;
+  socklen_t addr_len = sizeof(addr);
+  if (getsockname(sockfd, (struct sockaddr *)&addr, &addr_len) == -1) {
+    error = Status::FromErrorString(
```

Comment:
Please remove the `lldb-platform` reference from DomainSocket implementation.

Reply:
maybe `return GetLastError().takeError()`